crates/holochain/src/conductor/conductor.rs

```diff
@@ -1126,18 +1126,44 @@ mod builder {
     }
 }
 
+#[instrument(skip(p2p_evt, handle))]
 async fn p2p_event_task(
     mut p2p_evt: holochain_p2p::event::HolochainP2pEventReceiver,
     handle: ConductorHandle,
 ) {
+    /// The number of events we allow to run in parallel before
+    /// starting to await on the join handles.
+    const NUM_PARALLEL_EVTS: usize = 100;
     use tokio::stream::StreamExt;
+    let (mut tx, rx) = tokio::sync::mpsc::channel(NUM_PARALLEL_EVTS);
+
+    // Task to await a buffer of event join handles
+    tokio::task::spawn(async move {
+        use futures::StreamExt;
+        let mut buffer = rx.buffer_unordered(NUM_PARALLEL_EVTS);
+        while let Some(r) = futures::StreamExt::next(&mut buffer).await {
+            if let Err(e) = r {
+                error!("Failed to join event task {:?}", e);
+            }
+        }
+    });
+
+    // Spawn event tasks
     while let Some(evt) = p2p_evt.next().await {
```

Comment:
I feel like this join task is not needed complexity... i.e. what's checking that *this* task is joined?

This whole `async  fn p2p_event_task` is already spawned into a tokio task... so how about we make use of the for_each_concurrent directly off the `p2p_evt` receiver
```suggestion
    while let Some(evt) = p2p_evt.next().await {
    p2p_evt.for_each_concurrent(
        /* limit */ NUM_PARALLEL_EVTS,
        move |evt| async move {
            // ... dispatch here
        },
    ).await;
```

Reply:
This would run the events concurrently on a single thread though instead of using multiple threads. Is that what we want?

Reply:
Is there a way we can use the for each on the join handles?

Reply:
Looks like we could do:
```rust
    p2p_evt.for_each_concurrent(
        /* limit */ NUM_PARALLEL_EVTS,
        move |evt| spawn(async move {
            // ... dispatch here
        }),
    ).await;
```
Or maybe that's what you meant?

Reply:
There's no CPU load in the dispatching, though - it's awaiting async tasks, so should be totally fine for them all to run on the same thread, right?

Reply:
It's not sending on a channel in dispatch though, that's just an async function that calls the cell directly.

Reply:
Right, but the code inside the cell that is run when dispatch is called has `await`s in it, right? So any individual parallel execution will pause and let others run until all are Pending in which case this whole task will be paused until any one of those sub-wakers are invoked.